core/common/src/main/java/alluxio/ClientContext.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio;
+
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.InstancedConfiguration;
+import alluxio.exception.status.AlluxioStatusException;
+import alluxio.util.ConfigurationUtils;
+
+import java.net.InetSocketAddress;
+
+import javax.annotation.Nullable;
+import javax.security.auth.Subject;
+
+/**
+ * A {@link ClientContext} contains information required and pertaining to making network
+ * connections and performing operations with remote Alluxio processes. The {@link ClientContext}
+ * should only contain the information which is necessary to make those connections.
+ *
+ * A {@link ClientContext} are not expensive objects to create, however it is important to be
+ * aware that if the configuration with which the instance is created does not have the cluster
+ * default configuration loaded that any new clients which use the context will need to load the
+ * cluster defaults upon connecting to the Alluxio master.
+ *
+ * Ideally only a single {@link ClientContext} should be needed when initializing an application.
+ * This will use as few network resources as possible.
+ */
+public class ClientContext {
+  private volatile AlluxioConfiguration mConf;
+  private final Subject mSubject;
+
+  /**
+   * A client context with information about the subject and configuration of the client.
+   *
+   * @param subject the security subject to use
+   * @param alluxioConf the {@link AlluxioConfiguration} to use. If null, the site property defaults
+   *     will be loaded
+   * @return A new client context with the specified properties and subject
```

Comment:
nit: javadoc `@param` and `@return` should start lowercase and not end in period
```
➜  alluxio git:(58f168483e) ✗ git diff upstream/master | grep '@return [A-Z]' | grep '+'
+   * @return A new client context with the specified properties and subject
+   * @return A copy of AlluxioProperties representing the configuration
+   * @return A builder for a {@link MasterClientContext}
+   * @return UFS configuration from a given alluxio configuration
+   * @return The set of Alluxio properties loaded from the site-properties file
+   * @return A new configuration representing the merged properties
+   * @return An configuration object containing the original configuration merged with cluster
+   * @return The proper channel type USER_NETWORK_NETTY_CHANNEL
```

Reply:
The periods are caught by checkstyle. If we really want to enforce starting in lower case it should be added to checkstyle. I'll fix these though

Reply:
agreed, we can add it after the major outstanding PRs get in